### PR TITLE
Allocate map for ManifestURLHeader

### PR DESCRIPTION
**Pull request by mtaufen**
Issue: #55189

Fixes an assignment to nil map.

/cc @yliaog

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -1,307 +0,0 @@
-// +build !ignore_autogenerated
```
Comment (enisoc) [inline cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go]: Was this intentional?

**Comment by mtaufen** [comment]
Wth... looks like regenerating files left an artifact...

File: cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go (hunk 1 above)
Comment (mtaufen) [reply cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go]: No, fixing now. Not sure why this was generated.

**Comment by mindprince** [comment]
/test pull-kubernetes-e2e-gke-device-plugin-gpu

Running a test on GKE to make sure it works now.

File: cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go (hunk 1 above)
Comment (mtaufen) [reply cmd/kubeadm/app/phases/etcd/spec/zz_generated.deepcopy.go]: https://github.com/kubernetes/kubernetes/issues/55500 for this file not being regenerated

**Comment by enisoc** [comment]
It looks like kubelet starts up now, so this PR should help. The cluster is still not coming up due to a separate issue, which we'll continue to track on #55189.

**Pull request merged**